src/main/java/school/faang/user_service/service/goal/GoalServiceImpl.java

```diff
@@ -120,15 +124,26 @@ public GoalDto update(Long id, UpdateGoalDto updateGoalDto) {
 
         goalMapper.update(goal, updateGoalDto);
 
+        Goal updatedGoal = goalRepository.save(goal);
+
         if (goal.getStatus() == GoalStatus.COMPLETED && goal.getSkillsToAchieve() != null && goal.getUsers() != null) {
-            for (Skill skill : goal.getSkillsToAchieve()) {
-                for (User user : goal.getUsers()) {
+            for (User user : goal.getUsers()) {
+                List<SkillFilterDto> skills = new ArrayList<>();

```

Comment:
```suggestion
                    goal.getUsers().forEach(user -> {
        List<SkillFilterDto> skills = mapSkillsToDtos(goal.getSkillsToAchieve(), user);
        userUpdateProducer.onUserUpdate(new UserAddSkills(user.getId(), skills));
    });

```
в идеале это должно быть похоже на это